tango/integrations/beaker/workspace.py

```diff
@@ -118,7 +119,16 @@ def step_starting(self, step: Step) -> None:
         self.locks[step] = lock
 
         step_info = self.step_info(step)
-        if step_info.state not in {StepState.INCOMPLETE, StepState.FAILED, StepState.UNCACHEABLE}:
+        if step_info.state == StepState.RUNNING:
+            # Since we've acquired the step lock we know this step can't be running
+            # elsewhere. But the step state can still say its running if the last
+            # run exited before this workspace had a chance to update the step info.
```

Comment:
The `LocalWorkspace` has a function that guesses the step state given what we know about locks, existing directories, etc. Maybe it would be helpful to steal from there.

Reply:
I forgot about that. But I guess this is a simplified version because I think there's less edge cases to check here.